.gitignore

```diff
@@ -164,3 +164,4 @@ kafka-zookeeper-openshift.yaml
 nats.yaml
 kinesis.yaml
 /kubeless
+vendor/
```

Comment:
you are missing a blank line here

Reply:
Fixed.

I am looking into solving the issue with lates Kubernetes, trying to update the outdated dependencies for `kubeless` and `kafka-trigger`. This is quite tricky though:
1. There are circular dependencies between main repo - `kubeless`, and supporting repos - e.g. `kafka-trigger`. Might not be required, but it would make life easier if we either:
    1. Extract the utils (or other shared) packages into separate repos; or
    1. Merge everything in one repository: personally I would go with this approach, since it will be easier to ensure compatibility
1. Solving compatibility problems between various `k8s.io/*`  packages is nightmare.

Reply:
> Fixed.
> 
> I am looking into solving the issue with lates Kubernetes, trying to update the outdated dependencies for `kubeless` and `kafka-trigger`. This is quite tricky though:
> 
> 1. There are circular dependencies between main repo - `kubeless`, and supporting repos - e.g. `kafka-trigger`. Might not be required, but it would make life easier if we either:
>    
>    1. Extract the utils (or other shared) packages into separate repos; or
>    2. Merge everything in one repository: personally I would go with this approach, since it will be easier to ensure compatibility

The problem is that not only some `utils` functions are shared, in the `kubeless` repo, the API definitions of the triggers are used (and vice versa). Splitting everything in different packages can be a lot of work (that's why the circular dependency exists now), I agree that would be ideal but not sure if it's worth it.

> 2. Solving compatibility problems between various `k8s.io/*`  packages is nightmare.

Yeah, agree but at least https://github.com/kubeless/http-trigger/pull/14 seems to be a working combination? (Didn't try personally yet)